### PR TITLE
Avoid use of JS `arguments` in export wrappers

### PR DESCRIPTION
**Pull request by sbc100**
This allows the wrappers to be implemented as arrow functions, which saves some code size, and also avoids creating garbage every time these functions are called (admittedly that should only happen once since they re-bind on first usage).

### CODE REVIEW

**Comment by sbc100** [comment]
Depends on #19809

File: src/deterministic.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ Module['thisProgram'] = 'thisProgram'; // for consistency between different buil
 
 function hashMemory(id) {
   var ret = 0;
-  var len = _sbrk();
```
Comment (dschuff) [inline src/deterministic.js:29]: Are these changes needed because of strict mode?

Comment (sbc100) [reply src/deterministic.js:29]: They are needed because closure now knows the actual arity of the native functions.   I could revert that part and re-introduce the closure annotation?   But I think I actually prefer closure warnings when we calling with the wrong number of args.. and the number of places where we save a couple of bytes by not passing these extra args was very low.

**Review by dschuff — APPROVED** [review]
